pallets/schemas/src/lib.rs

```diff
@@ -263,6 +283,41 @@ pub mod pallet {
 			Self::deposit_event(Event::SchemaMaxSizeChanged { max_size });
 			Ok(())
 		}
+
+		/// Adds a given schema to storage with additional settings available from `SchemaSetting`
+		/// # Arguments
+		/// * `origin` - The origin of the call
+		/// * `model` - The schema model
+		/// * `model_type` - The schema model type
+		/// * `payload_location` - The schema payload location
+		/// * `settings` - The bounded list of additional schema settings.
+		///
+		/// # Events
+		/// * [`Event::SchemaCreated`]
+		///
+		/// # Errors
+		/// * [`Error::LessThanMinSchemaModelBytes`] - The schema's length is less than the minimum schema length
+		/// * [`Error::ExceedsMaxSchemaModelBytes`] - The schema's length is greater than the maximum schema length
+		/// * [`Error::InvalidSchema`] - Schema is malformed in some way
+		/// * [`Error::SchemaCountOverflow`] - The schema count has exceeded its bounds
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::create_schema(model.len() as u32 + settings.len() as u32))]
+		pub fn create_schema_with_settings(
+			origin: OriginFor<T>,
+			model: BoundedVec<u8, T::SchemaModelMaxBytesBoundedVecLimit>,
+			model_type: ModelType,
+			payload_location: PayloadLocation,
+			settings: BoundedVec<SchemaSetting, T::MaxSchemaSettingsPerSchema>,
```

Comment:
That would break the metadata so we would rather create a new extrinsic then forcing downstream to update metadata if they arent using stateful related stuff when we upgrade the chain and also we have a clear visibility.

open to suggestion

Reply:
And keeping option then it just behaves like original one, keeping it simple